### DISCUSSION TITLE
How to get OPC address?

### DISCUSSION THREAD

**Discussion by bartmch** [FreeOpcUa/opcua-asyncio#1299] [General]
When I'm connected to the PC which is connected to PLC's and has OPC server installed, I can read and write values to the PLC's directly using:
```
from pylogix import PLC
with PLC() as comm:
    comm.IPAddress = '192.168.0.2'
    ret = comm.Read('DCP_LP1_EGM_001_C')
```

My understanding is that reading/writing from/to PLC's directly is not the preffered way. We'd need to connect to the OPC server. The issue is that I don't know where to find the "RSLinx Classic" URL (opc_url below) to connect to it the following way:
```
from asyncua import Client
opc_url = 'opc.tcp://localhost:4840/freeopcua/server/'
async with Client(url=opc_url) as client:
    while True:
        # Do something with client
        node = client.get_node('i=85')
        value = await node.read_value()
```

**Comment by schroeder-** [comment]
There is a diference between opc ua and opc (classic). This libary does only support OPC UA,
Most servers the url is : 'opc.tcp://ip_or_hostname:4840' 
For example if the ip is 192.168.100.5: 'opc.tcp://192.16.100.5:4840'
For example if the hostname is plc1: 'opc.tcp://plc1:4840'

**Reply by bartmch** [reply]
Thank you for your prompt response. What would you recommend to read/write data using opc classic (using Python)?

**Comment by bartmch** [comment]
This slightly connected to my original question:
I want to set up an OPC UA Server() using opcua-asyncio. I would then like to use the client to build some logic and control PLC's.
How can I map the following PLC config example values in my OPC UA Server?

Matrikon.OPC.AllenBradleyPLCs.1
GroupName | Group0
ItemName | HP_1:RETAIL_8A2N49:REAL:E_DELTT.VALUE
ItemPath |  
DataType | 4
ReadOnly | 0
Reserved | 0
UpdateRate | 1000
Reserved | 0
Value | 6.86000013
Quality | Good, non-specific
Timestamp | 08:36.1

Using MetriconOPC it is possible to add a node and explore the variables:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/10415569/ad0d9193-ac3e-4819-b273-5903c7d943f2)
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/10415569/eb1f14c9-a694-4569-8ccb-b8c2490529fe)

After searching through the example files and documentation, one option could be to export XML config and import it using Server().import_xml() as in [simple-client-server-xml](https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/simple-client-server-xml/server.py) example?
